transaction/multisig.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2017 The Alvalor Authors
+//
+// This file is part of Alvalor.
+//
+// Alvalor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Alvalor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with Alvalor.  If not, see <http://www.gnu.org/licenses/>.
+
+package transaction
+
+// Signature represents a multisig wrapper for transactions.
+type Signature struct {
+	Action Action   // the underlying action to be executed
+	Sigs1  [][]byte // the signatures that should correspond to the origin of the action
+	Fee    Fee      // the attached fee transaction
+	Sigs2  [][]byte // the signatures that should correspond to the origin of the fees
+}
```

Comment:
This struct allows to assign Action field instance of Fee type and Fee field instance of Fee type as well. This would mean to pay fee for fee :). Did I understand it correctly?